providers/src/airflow/providers/edge/cli/edge_command.py

```diff
@@ -102,6 +102,32 @@ def _get_sysinfo() -> dict:
 def _pid_file_path(pid_file: str | None) -> str:
     return cli_utils.setup_locations(process=EDGE_WORKER_PROCESS_NAME, pid=pid_file)[0]
 
+def _write_pid_to_pidfile(pid_file_path):
+    """Write PID file to disk."""
+    if pid_file_path.exists():
+        # Handle existing PID files on disk
+        logger.info("An existing PID file has been found: %s.", pid_file_path)
+        pid_stored_in_pid_file = read_pid_from_pidfile(pid_file_path)
+        if os.getpid() == pid_stored_in_pid_file:
+            # case 1: PID file writing was triggered before for this instance
+            logger.info("PID file belongs to this process. File not updated.")
```

Comment:
I think this is a "should not happen". Kind of assert.
To be safe we should also raise an exception in this case.